### ISSUE TITLE
Prerelease notification: v4.4.0 - orders will be created after successful payment

### ISSUE THREAD

**Issue by AmandaMollie** [mollie/PrestaShop#403]
We have an upcoming release scheduled for this week that will have a high impact change for our merchants. 

As of v4.4.0, the order creation logic in the Mollie Plugin for Prestashop will change.

**All PrestaShop orders will be created only after successful payment.** This change will be the default logic within the module and it will therefore not be possible to create orders within PrestaShop before successful payment. 

We know this is a highly requested change so we'd like to ask for feedback during the beta release, before upload to the Prestashop marketplace. Thank you in advance for your engagement!

**Comment by AmandaMollie** [comment]
FYI - @ditwo @wzzly @spiros333 @maikel726 @folkertsmit @jcvoyeux @petrovv77 @adekonl @fredoche1810 @pantoffelke @RVanDamage @jacob-v-dam @daaaad

**Comment by Maikel726** [comment]
Hi,

At the moment I have not come across any critical errors, only that in test mode of the Mollie Module you cannot put orders on status: cancelled, open etc because then the order will remain stuck in a loading screen. Status paid works so I think this is due to the logic of how the orders are now created.

**Comment by Saimis777** [comment]
Hi @Maikel726 ,

Maybe you could raise a separate issue regarding this case? And provide us full steps on how to reproduce the issue please?

Thank you!

--
Best regards,
Invertus Support team

**Comment by folkertsmit** [comment]
I put it in live store, seems fine thanks!
Met vriendelijke groet,

Folkert Smit


Op do 16 sep. 2021 om 08:34 schreef Saimis777 ***@***.***>:

**Comment by wzzly** [comment]
After install, with the next order, I got the follow message added to the order: 
Warning: the secure key is empty, check your payment account before validation

I looked it up, and it has something to do with the PaymentModule.php file somewhere in validateOrder method
Seems like payment succeeded, but why is this message added as a note to the order? It's also on the delivery slip.

**Comment by Saimis777** [comment]
Hey @wzzly ,

Thanks for reporting that!

We're checking it right now, will update you ASAP!

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
Got a big issue the klarna authorized are not creating any order.
No order has been found on the order page

**Comment by Spiros333** [comment]
i got the same 
Warning: the secure key is empty, check your payment account before validation

**Comment by folkertsmit** [comment]
I have seen a problem twice now, order has been paid but the shopping cart
did not become an order in the webshop.

[image: image.png]

[image: image.png]

As you can see the description is mol_a number


Met vriendelijke groet,

Folkert Smit


Op wo 22 sep. 2021 om 16:08 schreef Saimis777 ***@***.***>:

**Comment by Spiros333** [comment]
@Saimis777  on the invoice 
![image](https://user-images.githubusercontent.com/85503636/134362260-484661fd-6a72-423e-831d-8b34e8e2f908.png)

**Comment by Spiros333** [comment]
@folkertsmit yes, I had it too.
I don't understand why which every release its just a mess everytime.
Why we can't get an good working plugin

**Comment by Saimis777** [comment]
@folkertsmit , @Spiros333 , @wzzly,

Thank you all for reporting this, I've opened a new issue for this - https://github.com/mollie/PrestaShop/issues/414

Maybe you could post your steps of reproduction there? As well as environments you are using? I would really appreciate that!

--
Best regards,
Invertus Support team

**Comment by AmandaMollie** [comment]
Thank you all for your feedback! Please continue to raise issues as you work with the new order creation logic and we will address any concerns there.

**Comment by folkertsmit** [comment]
Update  4.4.1 made it a lot better.

What I noticed recently that some orders will change to order status
cancelled after the payment is done.(around 15 minutes after).


Met vriendelijke groet,

Folkert Smit


Op di 28 sep. 2021 om 09:35 schreef AmandaMollie ***@***.***>:
